src/ui/actions/app.ts

```diff
@@ -209,6 +209,8 @@ export function loadMouseTargets(): UIThunkAction {
 
 export function setBreakpointsFromClipboard(): UIThunkAction {
   return async (dispatch, getState, { ThreadFront }) => {
+    // TODO [FE-998] Figure out a new implementation that works with IndexedDB instead
```

Comment:
Seems like maybe we could accomplish this (and simplify the other code?) in two parts:
1. Add index db initialization to the "basic info" loading step that blocks the rest of the app from loading.
2. Copy values into React state (so we can read/write them sync, in-memory) and just async update the DB.

Seems like that would enable us to undo a few of the awkward ref-copying, `[] | undefined` types logic in other places? am I missing something?

Reply:
This line here is also separate from the "React+IDB" question.

My first attempt at the "copy points to clipboard / paste points from clipboard" functionality relied on the fact that points were synced via `localStorage`, which has the `"storage"` event that fires whenever a value is set.

IndexedDB doesn't have any such event.  There's no way to know when it's been changed from somewhere else.

The "paste points from clipboard" feature actually didn't entirely work 100% right, so it's been sorta broken ever since I added it.

The change from storing points in `localStorage` to IDB further means that this implementation is not a viable approach.

So, I'm also turning this off until it can be rethought.

Reply:
Sounds fair.